aiidalab_widgets_base/aiida_visualizers.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import print_function
+import os
+
+import ipywidgets as ipw
+from aiida import load_dbenv, is_dbenv_loaded
+from aiida.backends import settings
+if not is_dbenv_loaded():
+    load_dbenv(profile=settings.AIIDADB_PROFILE)
+
+class ParameterDataVisualizer(ipw.HTML):
+    """Basic visualizer class for ParameterData object"""
+    def __init__(self, parameter, downloadable=True, **kwargs):
+        import pandas as pd
+        import base64
+        self.value = '''
+        <style>
+            .df tbody tr:nth-child(even) { background-color: lightgray;}
+            td { min-width: 300px; text-align: center; }
+        </style>
+        '''
+        pd.set_option('max_colwidth', 40)
+        df = pd.DataFrame([(key, value) for key, value
+                           in sorted(parameter.get_dict().items())
+                          ], columns=['Key', 'Value'])
+        self.value += '<center>' + df.to_html(classes='df', index=False) + '</center>'
+        if downloadable:
+            payload = base64.b64encode(df.to_csv().encode()).decode()
+            fname = '{}.csv'.format(parameter.pk)
+            to_add = """Download table in csv format: <a download="{filename}"
+            href="data:text/csv;base64,{payload}" target="_blank">{title}</a>"""
+            self.value += to_add.format(filename=fname, payload=payload,title=fname)
+        super(ParameterDataVisualizer, self).__init__(**kwargs)
+
+class StructureDataVisualizer(ipw.VBox):
+    """Basic visualizer class for StructureData object"""
+    def __init__(self, structure, downloadable=True, **kwargs):
+        import nglview
+        self._structure = structure
+        viewer = nglview.NGLWidget()
+        viewer.add_component(nglview.ASEStructure(self._structure.get_ase()))  # adds ball+stick
+        viewer.add_unitcell()
+        children = [viewer]
+        if downloadable:
+            self.file_format = ipw.Dropdown(
+                options=['xyz', 'cif', 'json'],
```

Comment:
what is this `json` format?
Is it some ASE-internal stuff?

Reply:
Yes, contains all the information that is present in ASE `Atoms` object. [Here](https://wiki.fysik.dtu.dk/ase/_modules/ase/io/formats.html) they just say "ASE JSON database file".

Might be useful if the structure is going to be converted into ASE again.

Reply:
Ok. Have you ever used it?
I believe most people won't know what "json" stands for here.
To keep things simple, perhaps better remove it.

Reply:
ok